wandb/sdk/launch/create_job.py

```diff
@@ -0,0 +1,430 @@
+        aliases (Optional[List[str]]): Aliases for the job.
+        runtime (Optional[str]): Python runtime of the job, like 3.9.
+        entrypoint (Optional[str]): Entrypoint of the job.
+        git_hash (Optional[str]): Git hash of a specific commit, when using git type jobs.
+
+
+    Returns:
+        Optional[Artifact]: The artifact created by the job, the action (for printing), and job aliases.
+                            None if job creation failed.
+
+    Example:
+        ```python
+        artifact_job = wandb.create_job(
+            job_type="code",
+            path=".",
+            entity="wandb",
+            project="jobs",
+            name="my-train-job",
+            description="My training job",
+            aliases=["train"],
+            runtime="3.9",
+            entrypoint="train.py",
+        )
+        # then, use you newly created job
+        artifact_job.call()
+        ```
+    """
+    api = Api()
+
+    artifact_job, _action, _aliases = _create_job(
+        api,
+        job_type,
+        path,
+        entity,
+        project,
+        name,
+        description,
+        aliases,
+        runtime,
+        entrypoint,
+        git_hash,
+    )
+
+    return artifact_job
+
+
+def _create_job(
+    api: Api,
+    job_type: str,
+    path: str,
+    entity: Optional[str] = None,
+    project: Optional[str] = None,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    aliases: Optional[List[str]] = None,
+    runtime: Optional[str] = None,
+    entrypoint: Optional[str] = None,
+    git_hash: Optional[str] = None,
+) -> Tuple[Optional[Artifact], str, List[str]]:
+    aliases = aliases or []
+    tempdir = tempfile.TemporaryDirectory()
+    metadata = {"_partial": "v0"}
+    requirements: List[str] = []
+
+    if job_type == "git":
+        repo_metadata = _create_repo_metadata(
+            path=path,
+            tempdir=tempdir.name,
+            entrypoint=entrypoint,
+            git_hash=git_hash,
+            runtime=runtime,
+        )
+        if not repo_metadata:
+            tempdir.cleanup()  # otherwise git can pollute
+            return None, "", []
+        metadata.update(repo_metadata)
+    elif job_type == "code":
+        path, entrypoint = _handle_artifact_entrypoint(path, entrypoint)
+        if not entrypoint:
+            wandb.termerror(
+                "Artifact jobs must have an entrypoint, either included in the path or specified with -E"
+            )
+            return None, "", []
+
+        artifact_metadata, requirements = _create_artifact_metadata(
+            path=path, entrypoint=entrypoint, runtime=runtime
+        )
+        if not artifact_metadata:
+            return None, "", []
+        metadata.update(artifact_metadata)
+    elif job_type == "image":
+        if runtime:
+            wandb.termwarn("Runtime is not supported for image jobs, ignoring runtime")
+        if entrypoint:
+            wandb.termwarn(
```

Comment:
nonblocking: This feels wrong, we'll want to support setting the entrypoint in image jobs

Reply:
right, its just not in the ImageSource, and wasn't quite sure why.... Easily added... @KyleGoyette

Reply:
Don't add it right now, image jobs dont support setting an entrypoint

Reply:
TODO